source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -63,7 +67,7 @@ class ExtProcLoggingInfo : public Envoy::StreamInfo::FilterState::Object {
     const Grpc::Status::GrpcStatus status_;
     const ProcessorState::CallbackState callback_state_;
   };
-  using GrpcCalls = std::vector<GrpcCall>;
+  using GrpcCalls = std::list<GrpcCall>;
```

Comment:
I think `deque` is a better choice than list

It provides random access that list doesn't and it has O(1) add and remove at both ends like list. I think add/remove in middle is not needed as here was a vector anyways.
Also I think deque has better performance (like memory locality/ cache) as it allocates contiguous memory while the list is a node-based container

Reply:
Understand that. However which one is more memory efficient?

Reply:
Compared with list, I believe is deque. 

As I mentioned above list is node-based container. each node is separately allocated. So the traversal operations of list will chase pointers to a new memory that results in a cache miss in most cases.  Moreover, changing from `vector` to `list` disable the random access ability it originally has.

If we want to maximize performance we probably should look for customized circular buffer. From a quick search, i found `SliceDeque` or `QuicheCircularDeque`. I am not sure if we really need them here.

Reply:
In that case, how about let's just used std::deque as it is already used in ext_proc filter here:
https://github.com/envoyproxy/envoy/blob/64e334de55386060e3124a0e9471e5b60804a4c0/source/extensions/filters/http/ext_proc/processor_state.h#L52

IIUC, QuicheCircularDeque is not a Envoy utility, and SliceDeque is but had not been used in Envoy.